app/uk/gov/hmrc/apinotificationpull/controllers/EnhancedNotificationsController.scala

```diff
@@ -47,8 +47,22 @@ class EnhancedNotificationsController @Inject()(enhancedApiNotificationQueueServ
 
   def pulled(notificationId: String): Action[AnyContent] = get(notificationId, Pulled, badRequestPulledText)
 
+  def pulledList(): Action[AnyContent] = getList(Pulled)
+
   def unpulled(notificationId: String): Action[AnyContent] = get(notificationId, Unpulled, badRequestUnpulledText)
 
+  def unpulledList(): Action[AnyContent] = getList(Unpulled)
+
+  private def getList(notificationStatus: NotificationStatus.Value): Action[AnyContent] =
+  (headerValidator.validateAcceptHeader andThen headerValidator.validateXClientIdHeader).async { implicit request =>
+
+    implicit val hc: HeaderCarrier = buildHeaderCarrier(request)
+
+    enhancedApiNotificationQueueService.getAllNotificationsBy(notificationStatus).map { notifications =>
+        Ok(enhancedXmlBuilder.toXml(notifications, notificationStatus)).as(XML)
+    } recover recovery
+  }
+
```

Comment:
what we are doing with 
```scala
          case Right(n) => Result(
            header = ResponseHeader(OK),
            body = HttpEntity.Strict(ByteString(n.payload), n.headers.get(CONTENT_TYPE)))
            .withHeaders(n.headers.toSeq: _*)
          case Left(nfe: NotFoundException) =>
            logger.info(s"Notification not found for id: $notificationId", nfe)
            ErrorNotFound.XmlResult
          case Left(bre: BadRequestException) =>
            logger.info(s"$badRequestText for id: $notificationId", bre)
            errorBadRequest(badRequestText).XmlResult
          case Left(e) =>
            logger.error(s"Internal server error for notification id: $notificationId", e)
            ErrorInternalServerError.XmlResult
        }.recover(recovery)
```
is packing an exception as a value/model in the connector, then decoding/unpacking it here as an error result.
We might as well return an error type of `ErrorResponse` from our connector, and also do the error logging there. Then in the controller all we need to do when we get a `Left` is `errorResult.Xml` to get the error response